src/domain/session/room/UnknownRoomViewModel.js

```diff
@@ -53,6 +61,38 @@ export class UnknownRoomViewModel extends ViewModel {
     }
 
     get kind() {
-        return "unknown";
+        return this._worldReadable ? "worldReadableRoom" : "unknown";
+    }
+
+    get timelineViewModel() {
+        return this._timelineVM;
+    }
+
+    avatarUrl(size) {
+        return getAvatarHttpUrl(this._room.avatarUrl, size, this.platform, this._room.mediaRepository);
+    }
+
+    async load() {
+        if (!this._worldReadable) {
+            return;
+        }
+        try {
+            this._room = await this._session.loadWorldReadableRoom(this.roomIdOrAlias);
+            const timeline = await this._room.openTimeline();
```

Comment:
Interesting ... how does this work if you don't already have a fragment of the timeline stored? What token do you end up calling /messages with?

Reply:
Actually `loadWorldReadableRoom()` above it does all of that - creates the Room object instance, fetch events from messages endpoint & load it all. You will see that here - https://github.com/Automattic/hydrogen-web/pull/3/files#diff-25d3af8c52953975b54837c62ef3195e1054638339aa9b1a9c1d6bc267cdadc8R962